fix: deprecate use of the favicon config parameter (#1225)
pydata/pydata-sphinx-theme: docs/user_guide/branding.rst

Add favicons
============

.. deprecated:: 0.15

>    The support for complex and multiple favicons will be dropped in version 0.15. Instead you can use the `sphinx-favicon <https://sphinx-favicon.readthedocs.io/en/stable/>`__ extention that provide the same functionalities using more flexible parameters.

Reviewer: ```suggestion
   Support for complex and multiple favicons will be dropped in version 0.15. Instead use the `sphinx-favicon <https://sphinx-favicon.readthedocs.io/en/stable/>`__ extension. It provides the same functionality using more flexible parameters.
```
